tests/graphicsItems/test_ImageItem.py

```diff
@@ -185,6 +185,45 @@ def test_ImageItem_axisorder():
         test_ImageItem(transpose=True)
     finally:
         pg.setConfigOptions(imageAxisOrder=origMode)
+        
+def test_setRect():    
+    def transforms_are_the_same(tr1, tr2):
+        dic = { # there seems to be no easy way to get the matrix in one call:
+            'tr11': ( tr1.m11(), tr2.m11() ),
+            'tr12': ( tr1.m12(), tr2.m12() ),
+            'tr13': ( tr1.m13(), tr2.m13() ),
+            'tr21': ( tr1.m21(), tr2.m21() ),
+            'tr22': ( tr1.m22(), tr2.m22() ),
+            'tr23': ( tr1.m23(), tr2.m23() ),
+            'tr31': ( tr1.m31(), tr2.m31() ),
+            'tr32': ( tr1.m32(), tr2.m32() ),
+            'tr33': ( tr1.m33(), tr2.m33() )
+        }
+        for key, values in dic.items():
+            val1, val2 = values
+            if val1 != val2:
+                print(key,'mismatch:', val1,'!=',val2 )
```

Comment:
This print-statement should probably be moved to the assert statement that calls this function.  I'm trying to think of a good way to embed this; only other alternative I can propose is to have this function handle the assert call as well; and have this print-statement baked in there.

Reply:
If you _really_ want to be slick about it, you can tell pytest how to display failed asserts for these types (totally not a requirement for this PR, just sharing)

https://docs.pytest.org/en/6.2.x/assert.html#defining-your-own-explanation-for-failed-assertions

Reply:
At the moment, we don't seem to test transform matrices a lot, so it is probably easier to keep everything together in the test function.
I changed the helper function to be `assert_equal_transforms()`.
It now checks all the elements and passes the assertion only when they are all the same. If they are not, it raises "AssertionError: Matrix element mismatch" with additional lines listing the mismatched matrix elements. These additional lines do not appear in the summary, but are captured in the detailed output:
```
E       AssertionError: Matrix element mismatch
E         tr11: 6.0 != 2.0
E         tr22: 16.0 != 4.0
E         tr31: -6.0 != -2.0
E         tr32: -16.0 != -4.0
E
E       assert False
```
That's the most helpful idea I currently have for providing some diagnostic output.